src/database/message-table.c

```diff
@@ -1282,10 +1310,8 @@ void logg_rate_limit_message(const char *clientIP, const unsigned int rate_limit
 	log_info("%s", buf);
 
 	// Log to database
-	const int rowid = add_message(RATE_LIMIT_MESSAGE, clientIP, config.dns.rateLimit.count.v.ui, config.dns.rateLimit.interval.v.ui, turnaround);
+	add_message(RATE_LIMIT_MESSAGE, clientIP, config.dns.rateLimit.count.v.ui, config.dns.rateLimit.interval.v.ui, turnaround);
 
-	if(rowid == -1)
```

Comment:
Why did your remove the check for the return code for all `add_message()` calls throughout the file?

Reply:
Maybe I'm wrong, but I think in case of failure (`rowid == -1`) the reason for failure is already logged inside `_add_message()` function. 
The message "Failed to add message to database" will be redundant.

Reply:
You are not wrong. It is to avoid printing warnings multiple times. I have checked that the innermost function already always reports an error in any case that can return an error back to the caller.

before
```
add_message(type=13, message=Failed to adjust time during NTP sync: Insufficient permissions) - SQL error step DELETE: attempt to write a readonly database
Error while trying to close database: database is locked
log_ntp_message(): Failed to add message to database
```

now
```
Cannot open database in read-write mode
```

(both taken from comments above)

Just seems cleaner to me.